Source/Scene/Globe.js

```diff
@@ -424,10 +426,44 @@ define([
         }
 
         var ellipsoid = this._surface._tileProvider.tilingScheme.ellipsoid;
-        var cartesian = ellipsoid.cartographicToCartesian(cartographic, scratchGetHeightCartesian);
+
+        //cartesian has to be on the ellipsoid surface for `ellipsoid.geodeticSurfaceNormal`
+        var cartesian = Cartesian3.fromRadians(cartographic.longitude, cartographic.latitude, 0.0, ellipsoid, scratchGetHeightCartesian);
 
         var ray = scratchGetHeightRay;
-        Cartesian3.normalize(cartesian, ray.direction);
+        var surfaceNormal = ellipsoid.geodeticSurfaceNormal(cartesian, ray.direction);
+
+        // compute origin point
+
+        // Try to find the intersection point between the surface normal and z-axis.
+        // Ellipsoid is a surface of revolution, so surface normal intersects the rotation axis (z-axis)
+
+        // compute the magnitude required to bring surface normal to x=0, y=0, from cartesian
+        var magnitude;
+
+        // avoid dividing by zero
+        if (Math.abs(surfaceNormal.x) > CesiumMath.EPSILON16){
+            magnitude = cartesian.x / surfaceNormal.x;
+        } else if (Math.abs(surfaceNormal.y) > CesiumMath.EPSILON16){
+            magnitude = cartesian.y / surfaceNormal.y;
+        } else if (Math.abs(surfaceNormal.z) > CesiumMath.EPSILON16){ //surface normal is (0,0,1) | (0,0,-1) | (0,0,0)
+            magnitude = cartesian.z / surfaceNormal.z;
+        } else { //(0,0,0), just for case
+            magnitude = 0;
+        }
+
+        var vectorToMinimumPoint = Cartesian3.multiplyByScalar(surfaceNormal, magnitude, scratchGetHeightIntersection);
+        Cartesian3.subtract(cartesian, vectorToMinimumPoint, ray.origin);
+
+        // Theoretically, the intersection point can be outside the ellipsoid, so we have to check if the result's 'z' is inside the ellipsoid (with some buffer)
+        if (Math.abs(ray.origin.z) >= ellipsoid.radii.z -11500.0){
```

Comment:
Instead of hard coding `-11500.0` here and below, can this use some percentage of [ellipsoid.radii.maximumComponent](https://github.com/AnalyticalGraphicsInc/cesium/blob/master/Source/Core/Cartesian3.js#L280)?  Or does the buffer need to be this big?  Instead, can it be a much smaller epsilon?